astronomer/providers/apache/livy/example_dags/example_livy.py

```diff
@@ -3,38 +3,321 @@
+    # Connect/ssh to an instance
+    cluster_response_master_public_dns = task_instance.xcom_pull(
+        key="cluster_response_master_public_dns", task_ids=["describe_created_cluster"]
+    )[0]
+    try:
+        client.connect(hostname=cluster_response_master_public_dns, username=kwargs["username"], pkey=key)
+
+        # Execute a command(cmd) after connecting/ssh to an instance
+        for command in kwargs["command"]:
+            stdin, stdout, stderr = client.exec_command(command)
+            stdout.read()
+
+        # close the client connection once the job is done
+        client.close()
+    except Exception as exc:
+        raise Exception("Got an exception as %s.", str(exc))
+
+
+def get_cluster_details(task_instance: Any) -> None:
+    """
+    Fetches the cluster details and stores EmrManagedMasterSecurityGroup and
+    MasterPublicDnsName in the XCOM.
+    """
+    client = boto3.client("emr", **AWS_S3_CREDS)
+    response = client.describe_cluster(
+        ClusterId=str(task_instance.xcom_pull(key="return_value", task_ids=["cluster_creator"])[0])
+    )
+    while (
+        "MasterPublicDnsName" not in response["Cluster"]
+        and response["Cluster"]["Status"]["State"] != "WAITING"
+    ):
+        logging.info("wait for 11 minutes to get the MasterPublicDnsName")
+        time.sleep(660)
+        response = client.describe_cluster(
+            ClusterId=str(task_instance.xcom_pull(key="return_value", task_ids=["cluster_creator"])[0])
+        )
+        logging.info("current response from ams emr: %s", str(response))
+    task_instance.xcom_push(
+        key="cluster_response_master_public_dns", value=response["Cluster"]["MasterPublicDnsName"]
+    )
+    task_instance.xcom_push(
+        key="cluster_response_master_security_group",
+        value=response["Cluster"]["Ec2InstanceAttributes"]["EmrManagedMasterSecurityGroup"],
+    )
+
 
 with DAG(
     dag_id="example_livy_operator",
-    default_args={"args": [10]},
     schedule_interval=None,
     start_date=datetime(2021, 1, 1),
     catchup=False,
     tags=["example", "async", "deferrable", "LivyOperatorAsync"],
 ) as dag:
+    # [START howto_create_key_pair_file]
+    create_key_pair_file = PythonOperator(
+        task_id="create_key_pair_file",
+        python_callable=create_key_pair,
+    )
+    # [END howto_create_key_pair_file]
+
+    # [START howto_operator_emr_create_job_flow]
+    cluster_creator = EmrCreateJobFlowOperator(
+        task_id="cluster_creator",
+        job_flow_overrides=JOB_FLOW_OVERRIDES,
+    )
+    # [END howto_operator_emr_create_job_flow]
+
+    # [START describe_created_cluster]
+    describe_created_cluster = PythonOperator(
+        task_id="describe_created_cluster", python_callable=get_cluster_details
+    )
+    # [END describe_created_cluster]
 
-    # [START create_livy]
+    # [START add_example_pi_file_in_hdfs]
+    ssh_and_copy_pifile_to_hdfs = PythonOperator(
```

Comment:
Could we have used [`SFTPOperator`](https://airflow.apache.org/docs/apache-airflow/1.10.12/_api/airflow/contrib/operators/sftp_operator/index.html) to transfer the pem file?

Reply:
will check and include in the next PR if thats ok